// @openmoviedb/kinopoiskdev_client/dist/main.js
function $parcel$exportWildcard(dest, source) {
  Object.keys(source).forEach(function(key) {
    if (key === 'default' || key === '__esModule' || dest.hasOwnProperty(key)) {
      return;
    }

    Object.defineProperty(dest, key, {
      enumerable: true,
      get: function get() {
        return source[key];
      }
    });
  });

  return dest;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $781ead698aa28b64$exports = {};

$parcel$export($781ead698aa28b64$exports, "KinopoiskDev", () => $781ead698aa28b64$export$1af66cb2cba0b3db);
var $df83c35e0194227c$exports = {};

$parcel$export($df83c35e0194227c$exports, "VERSION", () => $df83c35e0194227c$export$a4ad2735b021c132);
var $df83c35e0194227c$export$a4ad2735b021c132;
(function(VERSION) {
    VERSION["V1"] = "v1";
    VERSION["V1_1"] = "v1_2";
    VERSION["V1_2"] = "v1.2";
    VERSION["V1_3"] = "v1.3";
})($df83c35e0194227c$export$a4ad2735b021c132 || ($df83c35e0194227c$export$a4ad2735b021c132 = {}));


class $d832fe52df85a399$export$119dceb9b5803d83 {
    request;
    constructor(request){
        this.request = request;
    }
    async getByFilters(filters) {
        return await this.request.get((0, $df83c35e0194227c$export$a4ad2735b021c132).V1, `/review`, filters);
    }
}



class $53121b0f64a6db72$export$6c1ff094fe2bb12f {
    request;
    constructor(request){
        this.request = request;
    }
    async getByFilters(filters) {
        return await this.request.get((0, $df83c35e0194227c$export$a4ad2735b021c132).V1, `/season`, filters);
    }
}



class $db26abee0fa06c22$export$126adf3087e9debe {
    request;
    constructor(request){
        this.request = request;
    }
    async getById(id) {
        return await this.request.get((0, $df83c35e0194227c$export$a4ad2735b021c132).V1_3, `/movie/${id}`);
    }
    async getRandom() {
        return await this.request.get((0, $df83c35e0194227c$export$a4ad2735b021c132).V1_3, `/movie/random`);
    }
    async getByFilters(filters) {
        return await this.request.get((0, $df83c35e0194227c$export$a4ad2735b021c132).V1_3, `/movie`, filters);
    }
    async getBySearchQuery(filters) {
        return await this.request.get((0, $df83c35e0194227c$export$a4ad2735b021c132).V1_2, `/movie/search`, filters);
    }
    async getAwardsByFilters(filters, paginationParams) {
        return await this.request.get((0, $df83c35e0194227c$export$a4ad2735b021c132).V1_1, `/movie/awards`, filters);
    }
    getPossibleValuesByField(field) {
        return this.request.get((0, $df83c35e0194227c$export$a4ad2735b021c132).V1, `/movie/possible-values-by-field`, {
            field: field
        });
    }
}



class $1698a013b910504b$export$77b570809fa8c4b2 {
    request;
    constructor(request){
        this.request = request;
    }
    async getById(id) {
        return await this.request.get((0, $df83c35e0194227c$export$a4ad2735b021c132).V1, `/person/${id}`);
    }
    async getByFilters(filters) {
        return await this.request.get((0, $df83c35e0194227c$export$a4ad2735b021c132).V1, `/person`, filters);
    }
    async getBySearchQuery(filters) {
        return await this.request.get((0, $df83c35e0194227c$export$a4ad2735b021c132).V1_2, `/person/search`, filters);
    }
    async getAwardsByFilters(filters, paginationParams) {
        return await this.request.get((0, $df83c35e0194227c$export$a4ad2735b021c132).V1_1, `/movie/awards`, filters);
    }
}



class $4b1405a957cd9194$export$ea2bb5c7a344080e {
    request;
    constructor(request){
        this.request = request;
    }
    async getByFilters(filters) {
        return await this.request.get((0, $df83c35e0194227c$export$a4ad2735b021c132).V1, `/keyword`, filters);
    }
}



class $0df26f06a897adc7$export$890b60d5b725593 {
    request;
    constructor(request){
        this.request = request;
    }
    async getByFilters(filters) {
        return await this.request.get((0, $df83c35e0194227c$export$a4ad2735b021c132).V1, `/image`, filters);
    }
}



class $4dff5eafb332a955$export$11b871f7fddb2f94 {
    request;
    constructor(request){
        this.request = request;
    }
    async getById(id) {
        return await this.request.get((0, $df83c35e0194227c$export$a4ad2735b021c132).V1, `/studio/${id}`);
    }
    async getByFilters(filters) {
        return await this.request.get((0, $df83c35e0194227c$export$a4ad2735b021c132).V1, `/studio`, filters);
    }
}


class $778fdffd28ed3b16$export$93ae719a88c5bb5f {
    API_KEY;
    API_URL;
    constructor(API_KEY, API_URL){
        this.API_KEY = API_KEY;
        this.API_URL = API_URL;
    }
    queryParams(params) {
        const urlSearchParams = new URLSearchParams();
        for(const key in params){
            if (Array.isArray(params[key])) for (const value of params[key])urlSearchParams.append(key, value);
            else urlSearchParams.append(key, params[key]);
        }
        return urlSearchParams.toString();
    }
    async get(version, path, params) {
        try {
            const response = await fetch(`${this.API_URL}/${version}${path}?${this.queryParams(params)}`, {
                headers: {
                    "X-API-KEY": this.API_KEY
                }
            });
            // Если статус HTTP-ответа не 200, выбрасывается ошибка
            if (!response.ok) {
                const error = await response.json();
                throw new Error(JSON.stringify({
                    status: response.status,
                    error: error.error,
                    message: error.message
                }));
            }
            return {
                data: await response.json(),
                statusCode: response.status,
                error: null,
                message: null
            };
        } catch (error) {
            const { status: status, message: message, error: err } = JSON.parse(error.message);
            console.log("There was an error with your request:", error);
            return {
                data: null,
                statusCode: status,
                error: err,
                message: message
            };
        }
    }
}


class $781ead698aa28b64$export$1af66cb2cba0b3db {
    API_KEY;
    API_URL;
    movie;
    season;
    person;
    review;
    keyword;
    studio;
    image;
    request;
    constructor(API_KEY, API_URL = "https://api.kinopoisk.dev"){
        this.API_KEY = API_KEY;
        this.API_URL = API_URL;
        this.request = new (0, $778fdffd28ed3b16$export$93ae719a88c5bb5f)(this.API_KEY, this.API_URL);
        this.movie = new (0, $db26abee0fa06c22$export$126adf3087e9debe)(this.request);
        this.season = new (0, $53121b0f64a6db72$export$6c1ff094fe2bb12f)(this.request);
        this.person = new (0, $1698a013b910504b$export$77b570809fa8c4b2)(this.request);
        this.keyword = new (0, $4b1405a957cd9194$export$ea2bb5c7a344080e)(this.request);
        this.studio = new (0, $4dff5eafb332a955$export$11b871f7fddb2f94)(this.request);
        this.image = new (0, $0df26f06a897adc7$export$890b60d5b725593)(this.request);
        this.review = new (0, $d832fe52df85a399$export$119dceb9b5803d83)(this.request);
    }
}


var $7b13cf01bb3ea549$exports = {};
var $64ee512f47691b63$exports = {};

$parcel$export($64ee512f47691b63$exports, "QueryBuilder", () => $64ee512f47691b63$export$d24bb501ea7ded0b);
$parcel$export($64ee512f47691b63$exports, "MovieQueryBuilder", () => $64ee512f47691b63$export$819b0b6b1f83fa2a);
$parcel$export($64ee512f47691b63$exports, "PersonQueryBuilder", () => $64ee512f47691b63$export$ae3fb5834197d0a9);
$parcel$export($64ee512f47691b63$exports, "ReviewQueryBuilder", () => $64ee512f47691b63$export$8659e94a5c99ae98);
$parcel$export($64ee512f47691b63$exports, "SeasonQueryBuilder", () => $64ee512f47691b63$export$1828073978eaf824);
$parcel$export($64ee512f47691b63$exports, "StudioQueryBuilder", () => $64ee512f47691b63$export$efddcc2335f0c393);
$parcel$export($64ee512f47691b63$exports, "ImageQueryBuilder", () => $64ee512f47691b63$export$a8205ba3f55eed92);
class $8aee3cac8d273170$export$68f5e1453c188a82 {
    page;
    limit;
    constructor(page, limit){
        if (page && page < 1) throw new Error("Page must be greater than 0");
        if (limit && limit < 1) throw new Error("Limit must be greater than 0");
        if (limit && limit > 250) throw new Error("Limit must be less than or equal to 250");
        this.page = page || 1;
        this.limit = limit || 10;
    }
}


class $64ee512f47691b63$export$d24bb501ea7ded0b {
    params;
    constructor(){
        this.params = {};
    }
    select(fields) {
        this.params.selectFields = fields;
        return this;
    }
    sort(field, sortType) {
        if (!this.params.sortField) {
            this.params.sortField = [];
            this.params.sortType = [];
        }
        this.params.sortField.push(field);
        this.params.sortType.push(sortType);
        return this;
    }
    filterExact(field, value) {
        if (!this.params[field]) this.params[field] = [];
        this.params[field].push(value);
        return this;
    }
    filterRange(field, range) {
        this.params[field] = range.join("-");
        return this;
    }
    filterDateRange(field, range) {
        this.params[field] = range.map((date)=>date.toISOString().split("T")[0]).join("-");
        return this;
    }
    query(query) {
        this.params.query = query;
        return this;
    }
    paginate(page, limit) {
        const pagination = new (0, $8aee3cac8d273170$export$68f5e1453c188a82)(page, limit);
        this.params.page = pagination.page;
        this.params.limit = pagination.limit;
        return this;
    }
    build() {
        return this.params;
    }
}
class $64ee512f47691b63$export$819b0b6b1f83fa2a extends $64ee512f47691b63$export$d24bb501ea7ded0b {
}
class $64ee512f47691b63$export$ae3fb5834197d0a9 extends $64ee512f47691b63$export$d24bb501ea7ded0b {
}
class $64ee512f47691b63$export$8659e94a5c99ae98 extends $64ee512f47691b63$export$d24bb501ea7ded0b {
}
class $64ee512f47691b63$export$1828073978eaf824 extends $64ee512f47691b63$export$d24bb501ea7ded0b {
}
class $64ee512f47691b63$export$efddcc2335f0c393 extends $64ee512f47691b63$export$d24bb501ea7ded0b {
}
class $64ee512f47691b63$export$a8205ba3f55eed92 extends $64ee512f47691b63$export$d24bb501ea7ded0b {
}


$parcel$exportWildcard($7b13cf01bb3ea549$exports, $64ee512f47691b63$exports);


var $f4095d8f23ee8c4d$exports = {};
var $347178954a01dea6$exports = {};

$parcel$export($347178954a01dea6$exports, "SORT_TYPE", () => $347178954a01dea6$export$a40e5455e0e769de);
var $347178954a01dea6$export$a40e5455e0e769de;
(function(SORT_TYPE) {
    SORT_TYPE["ASC"] = "1";
    SORT_TYPE["DESC"] = "-1";
})($347178954a01dea6$export$a40e5455e0e769de || ($347178954a01dea6$export$a40e5455e0e769de = {}));


var $7a334a6fd85205e2$exports = {};

$parcel$export($7a334a6fd85205e2$exports, "SPECIAL_VALUE", () => $7a334a6fd85205e2$export$5286ee09df71cffc);
var $7a334a6fd85205e2$export$5286ee09df71cffc;
(function(SPECIAL_VALUE) {
    SPECIAL_VALUE["NOT_NULL"] = "!null";
})($7a334a6fd85205e2$export$5286ee09df71cffc || ($7a334a6fd85205e2$export$5286ee09df71cffc = {}));



$parcel$exportWildcard($f4095d8f23ee8c4d$exports, $347178954a01dea6$exports);
$parcel$exportWildcard($f4095d8f23ee8c4d$exports, $7a334a6fd85205e2$exports);
$parcel$exportWildcard($f4095d8f23ee8c4d$exports, $df83c35e0194227c$exports);


var $da39c0936ef66375$exports = {};
var $9d71295bb8991db6$exports = {};


var $bb267bac4f590fd8$exports = {};


var $51eb17df087c289f$exports = {};


var $a5e5e1439e435731$exports = {};


var $6d809023321425ba$exports = {};


var $b6198800830fc815$exports = {};


var $e22dca8c9df55148$exports = {};

$parcel$export($e22dca8c9df55148$exports, "Studio", () => $e22dca8c9df55148$export$83fb124b13439b08);
/**
 * @export
 * @namespace Studio
 */ var $e22dca8c9df55148$export$83fb124b13439b08;
(function(Studio) {
    /**
     * @export
     * @enum {string}
     */ let TypeEnum;
    (function(TypeEnum) {
        TypeEnum[TypeEnum["_0"] = "Производство"] = "_0";
        TypeEnum[TypeEnum["_1"] = "Спецэффекты"] = "_1";
        TypeEnum[TypeEnum["_2"] = "Прокат"] = "_2";
        TypeEnum[TypeEnum["_3"] = "Студия дубляжа"] = "_3";
    })(TypeEnum = Studio.TypeEnum || (Studio.TypeEnum = {}));
})($e22dca8c9df55148$export$83fb124b13439b08 || ($e22dca8c9df55148$export$83fb124b13439b08 = {}));


var $2deceb19604ee211$exports = {};
var $2937fd045a754fc7$exports = {};


var $f2f72605d797d0a9$exports = {};


var $8e4f7c074acb8cbf$exports = {};


var $ef3142cf1ef1a5f4$exports = {};


var $2035eab642e949a1$exports = {};


var $3507abb8ec29fc15$exports = {};


var $c4f36d0416fca157$exports = {};


$parcel$exportWildcard($2deceb19604ee211$exports, $2937fd045a754fc7$exports);
$parcel$exportWildcard($2deceb19604ee211$exports, $f2f72605d797d0a9$exports);
$parcel$exportWildcard($2deceb19604ee211$exports, $8e4f7c074acb8cbf$exports);
$parcel$exportWildcard($2deceb19604ee211$exports, $ef3142cf1ef1a5f4$exports);
$parcel$exportWildcard($2deceb19604ee211$exports, $2035eab642e949a1$exports);
$parcel$exportWildcard($2deceb19604ee211$exports, $3507abb8ec29fc15$exports);
$parcel$exportWildcard($2deceb19604ee211$exports, $c4f36d0416fca157$exports);


$parcel$exportWildcard($da39c0936ef66375$exports, $9d71295bb8991db6$exports);
$parcel$exportWildcard($da39c0936ef66375$exports, $bb267bac4f590fd8$exports);
$parcel$exportWildcard($da39c0936ef66375$exports, $51eb17df087c289f$exports);
$parcel$exportWildcard($da39c0936ef66375$exports, $a5e5e1439e435731$exports);
$parcel$exportWildcard($da39c0936ef66375$exports, $6d809023321425ba$exports);
$parcel$exportWildcard($da39c0936ef66375$exports, $b6198800830fc815$exports);
$parcel$exportWildcard($da39c0936ef66375$exports, $e22dca8c9df55148$exports);
$parcel$exportWildcard($da39c0936ef66375$exports, $2deceb19604ee211$exports);


var $77847526171ae7cb$exports = {};
var $51c8238cf6dae8a0$exports = {};


var $aec5e624723e2b98$exports = {};


var $436d32dda76838f0$exports = {};


var $f2a6f24546447701$exports = {};


var $cbd9a3e2c8e02e8a$exports = {};


var $57af2a862a2bc14c$exports = {};


var $6fadc5b8e36418e5$exports = {};


$parcel$exportWildcard($77847526171ae7cb$exports, $51c8238cf6dae8a0$exports);
$parcel$exportWildcard($77847526171ae7cb$exports, $aec5e624723e2b98$exports);
$parcel$exportWildcard($77847526171ae7cb$exports, $436d32dda76838f0$exports);
$parcel$exportWildcard($77847526171ae7cb$exports, $f2a6f24546447701$exports);
$parcel$exportWildcard($77847526171ae7cb$exports, $cbd9a3e2c8e02e8a$exports);
$parcel$exportWildcard($77847526171ae7cb$exports, $57af2a862a2bc14c$exports);
$parcel$exportWildcard($77847526171ae7cb$exports, $6fadc5b8e36418e5$exports);


$parcel$exportWildcard(module.exports, $7b13cf01bb3ea549$exports);
$parcel$exportWildcard(module.exports, $f4095d8f23ee8c4d$exports);
$parcel$exportWildcard(module.exports, $781ead698aa28b64$exports);
$parcel$exportWildcard(module.exports, $da39c0936ef66375$exports);
$parcel$exportWildcard(module.exports, $77847526171ae7cb$exports);
