// @openmoviedb/kinopoiskdev_client/dist/module.js
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
function $parcel$exportWildcard(dest, source) {
  Object.keys(source).forEach(function(key) {
    if (key === 'default' || key === '__esModule' || dest.hasOwnProperty(key)) {
      return;
    }

    Object.defineProperty(dest, key, {
      enumerable: true,
      get: function get() {
        return source[key];
      }
    });
  });

  return dest;
}
var $3c880564bcfe2be9$exports = {};

$parcel$export($3c880564bcfe2be9$exports, "KinopoiskDev", () => $3c880564bcfe2be9$export$1af66cb2cba0b3db);
var $3575132427cc3c5d$exports = {};

$parcel$export($3575132427cc3c5d$exports, "VERSION", () => $3575132427cc3c5d$export$a4ad2735b021c132);
var $3575132427cc3c5d$export$a4ad2735b021c132;
(function(VERSION) {
    VERSION["V1"] = "v1";
    VERSION["V1_1"] = "v1_2";
    VERSION["V1_2"] = "v1.2";
    VERSION["V1_3"] = "v1.3";
})($3575132427cc3c5d$export$a4ad2735b021c132 || ($3575132427cc3c5d$export$a4ad2735b021c132 = {}));


class $efc1e4dd1b90c575$export$119dceb9b5803d83 {
    request;
    constructor(request){
        this.request = request;
    }
    async getByFilters(filters) {
        return await this.request.get((0, $3575132427cc3c5d$export$a4ad2735b021c132).V1, `/review`, filters);
    }
}



class $53766d2d77f086f8$export$6c1ff094fe2bb12f {
    request;
    constructor(request){
        this.request = request;
    }
    async getByFilters(filters) {
        return await this.request.get((0, $3575132427cc3c5d$export$a4ad2735b021c132).V1, `/season`, filters);
    }
}



class $6ed2074377cce267$export$126adf3087e9debe {
    request;
    constructor(request){
        this.request = request;
    }
    async getById(id) {
        return await this.request.get((0, $3575132427cc3c5d$export$a4ad2735b021c132).V1_3, `/movie/${id}`);
    }
    async getRandom() {
        return await this.request.get((0, $3575132427cc3c5d$export$a4ad2735b021c132).V1_3, `/movie/random`);
    }
    async getByFilters(filters) {
        return await this.request.get((0, $3575132427cc3c5d$export$a4ad2735b021c132).V1_3, `/movie`, filters);
    }
    async getBySearchQuery(filters) {
        return await this.request.get((0, $3575132427cc3c5d$export$a4ad2735b021c132).V1_2, `/movie/search`, filters);
    }
    async getAwardsByFilters(filters, paginationParams) {
        return await this.request.get((0, $3575132427cc3c5d$export$a4ad2735b021c132).V1_1, `/movie/awards`, filters);
    }
    getPossibleValuesByField(field) {
        return this.request.get((0, $3575132427cc3c5d$export$a4ad2735b021c132).V1, `/movie/possible-values-by-field`, {
            field: field
        });
    }
}



class $47dad12ec229e9f7$export$77b570809fa8c4b2 {
    request;
    constructor(request){
        this.request = request;
    }
    async getById(id) {
        return await this.request.get((0, $3575132427cc3c5d$export$a4ad2735b021c132).V1, `/person/${id}`);
    }
    async getByFilters(filters) {
        return await this.request.get((0, $3575132427cc3c5d$export$a4ad2735b021c132).V1, `/person`, filters);
    }
    async getBySearchQuery(filters) {
        return await this.request.get((0, $3575132427cc3c5d$export$a4ad2735b021c132).V1_2, `/person/search`, filters);
    }
    async getAwardsByFilters(filters, paginationParams) {
        return await this.request.get((0, $3575132427cc3c5d$export$a4ad2735b021c132).V1_1, `/movie/awards`, filters);
    }
}



class $7fb2e498f013a095$export$ea2bb5c7a344080e {
    request;
    constructor(request){
        this.request = request;
    }
    async getByFilters(filters) {
        return await this.request.get((0, $3575132427cc3c5d$export$a4ad2735b021c132).V1, `/keyword`, filters);
    }
}



class $b4bf5ffeb5cd48b1$export$890b60d5b725593 {
    request;
    constructor(request){
        this.request = request;
    }
    async getByFilters(filters) {
        return await this.request.get((0, $3575132427cc3c5d$export$a4ad2735b021c132).V1, `/image`, filters);
    }
}



class $c71652c405502cca$export$11b871f7fddb2f94 {
    request;
    constructor(request){
        this.request = request;
    }
    async getById(id) {
        return await this.request.get((0, $3575132427cc3c5d$export$a4ad2735b021c132).V1, `/studio/${id}`);
    }
    async getByFilters(filters) {
        return await this.request.get((0, $3575132427cc3c5d$export$a4ad2735b021c132).V1, `/studio`, filters);
    }
}


class $8928faa2ed7a6b78$export$93ae719a88c5bb5f {
    API_KEY;
    API_URL;
    constructor(API_KEY, API_URL){
        this.API_KEY = API_KEY;
        this.API_URL = API_URL;
    }
    queryParams(params) {
        const urlSearchParams = new URLSearchParams();
        for(const key in params){
            if (Array.isArray(params[key])) for (const value of params[key])urlSearchParams.append(key, value);
            else urlSearchParams.append(key, params[key]);
        }
        return urlSearchParams.toString();
    }
    async get(version, path, params) {
        try {
            const response = await fetch(`${this.API_URL}/${version}${path}?${this.queryParams(params)}`, {
                headers: {
                    "X-API-KEY": this.API_KEY
                }
            });
            // Если статус HTTP-ответа не 200, выбрасывается ошибка
            if (!response.ok) {
                const error = await response.json();
                throw new Error(JSON.stringify({
                    status: response.status,
                    error: error.error,
                    message: error.message
                }));
            }
            return {
                data: await response.json(),
                statusCode: response.status,
                error: null,
                message: null
            };
        } catch (error) {
            const { status: status, message: message, error: err } = JSON.parse(error.message);
            console.log("There was an error with your request:", error);
            return {
                data: null,
                statusCode: status,
                error: err,
                message: message
            };
        }
    }
}


class $3c880564bcfe2be9$export$1af66cb2cba0b3db {
    API_KEY;
    API_URL;
    movie;
    season;
    person;
    review;
    keyword;
    studio;
    image;
    request;
    constructor(API_KEY, API_URL = "https://api.kinopoisk.dev"){
        this.API_KEY = API_KEY;
        this.API_URL = API_URL;
        this.request = new (0, $8928faa2ed7a6b78$export$93ae719a88c5bb5f)(this.API_KEY, this.API_URL);
        this.movie = new (0, $6ed2074377cce267$export$126adf3087e9debe)(this.request);
        this.season = new (0, $53766d2d77f086f8$export$6c1ff094fe2bb12f)(this.request);
        this.person = new (0, $47dad12ec229e9f7$export$77b570809fa8c4b2)(this.request);
        this.keyword = new (0, $7fb2e498f013a095$export$ea2bb5c7a344080e)(this.request);
        this.studio = new (0, $c71652c405502cca$export$11b871f7fddb2f94)(this.request);
        this.image = new (0, $b4bf5ffeb5cd48b1$export$890b60d5b725593)(this.request);
        this.review = new (0, $efc1e4dd1b90c575$export$119dceb9b5803d83)(this.request);
    }
}


var $a0c0be9502ba1dfc$exports = {};
var $21782052d57b2416$exports = {};

$parcel$export($21782052d57b2416$exports, "QueryBuilder", () => $21782052d57b2416$export$d24bb501ea7ded0b);
$parcel$export($21782052d57b2416$exports, "MovieQueryBuilder", () => $21782052d57b2416$export$819b0b6b1f83fa2a);
$parcel$export($21782052d57b2416$exports, "PersonQueryBuilder", () => $21782052d57b2416$export$ae3fb5834197d0a9);
$parcel$export($21782052d57b2416$exports, "ReviewQueryBuilder", () => $21782052d57b2416$export$8659e94a5c99ae98);
$parcel$export($21782052d57b2416$exports, "SeasonQueryBuilder", () => $21782052d57b2416$export$1828073978eaf824);
$parcel$export($21782052d57b2416$exports, "StudioQueryBuilder", () => $21782052d57b2416$export$efddcc2335f0c393);
$parcel$export($21782052d57b2416$exports, "ImageQueryBuilder", () => $21782052d57b2416$export$a8205ba3f55eed92);
class $510154db16c6d3d9$export$68f5e1453c188a82 {
    page;
    limit;
    constructor(page, limit){
        if (page && page < 1) throw new Error("Page must be greater than 0");
        if (limit && limit < 1) throw new Error("Limit must be greater than 0");
        if (limit && limit > 250) throw new Error("Limit must be less than or equal to 250");
        this.page = page || 1;
        this.limit = limit || 10;
    }
}


class $21782052d57b2416$export$d24bb501ea7ded0b {
    params;
    constructor(){
        this.params = {};
    }
    select(fields) {
        this.params.selectFields = fields;
        return this;
    }
    sort(field, sortType) {
        if (!this.params.sortField) {
            this.params.sortField = [];
            this.params.sortType = [];
        }
        this.params.sortField.push(field);
        this.params.sortType.push(sortType);
        return this;
    }
    filterExact(field, value) {
        if (!this.params[field]) this.params[field] = [];
        this.params[field].push(value);
        return this;
    }
    filterRange(field, range) {
        this.params[field] = range.join("-");
        return this;
    }
    filterDateRange(field, range) {
        this.params[field] = range.map((date)=>date.toISOString().split("T")[0]).join("-");
        return this;
    }
    query(query) {
        this.params.query = query;
        return this;
    }
    paginate(page, limit) {
        const pagination = new (0, $510154db16c6d3d9$export$68f5e1453c188a82)(page, limit);
        this.params.page = pagination.page;
        this.params.limit = pagination.limit;
        return this;
    }
    build() {
        return this.params;
    }
}
class $21782052d57b2416$export$819b0b6b1f83fa2a extends $21782052d57b2416$export$d24bb501ea7ded0b {
}
class $21782052d57b2416$export$ae3fb5834197d0a9 extends $21782052d57b2416$export$d24bb501ea7ded0b {
}
class $21782052d57b2416$export$8659e94a5c99ae98 extends $21782052d57b2416$export$d24bb501ea7ded0b {
}
class $21782052d57b2416$export$1828073978eaf824 extends $21782052d57b2416$export$d24bb501ea7ded0b {
}
class $21782052d57b2416$export$efddcc2335f0c393 extends $21782052d57b2416$export$d24bb501ea7ded0b {
}
class $21782052d57b2416$export$a8205ba3f55eed92 extends $21782052d57b2416$export$d24bb501ea7ded0b {
}


$parcel$exportWildcard($a0c0be9502ba1dfc$exports, $21782052d57b2416$exports);


var $d88c1fb2ace5e8db$exports = {};
var $6c79b96500ca857b$exports = {};

$parcel$export($6c79b96500ca857b$exports, "SORT_TYPE", () => $6c79b96500ca857b$export$a40e5455e0e769de);
var $6c79b96500ca857b$export$a40e5455e0e769de;
(function(SORT_TYPE) {
    SORT_TYPE["ASC"] = "1";
    SORT_TYPE["DESC"] = "-1";
})($6c79b96500ca857b$export$a40e5455e0e769de || ($6c79b96500ca857b$export$a40e5455e0e769de = {}));


var $dd8fe0960d14576e$exports = {};

$parcel$export($dd8fe0960d14576e$exports, "SPECIAL_VALUE", () => $dd8fe0960d14576e$export$5286ee09df71cffc);
var $dd8fe0960d14576e$export$5286ee09df71cffc;
(function(SPECIAL_VALUE) {
    SPECIAL_VALUE["NOT_NULL"] = "!null";
})($dd8fe0960d14576e$export$5286ee09df71cffc || ($dd8fe0960d14576e$export$5286ee09df71cffc = {}));



$parcel$exportWildcard($d88c1fb2ace5e8db$exports, $6c79b96500ca857b$exports);
$parcel$exportWildcard($d88c1fb2ace5e8db$exports, $dd8fe0960d14576e$exports);
$parcel$exportWildcard($d88c1fb2ace5e8db$exports, $3575132427cc3c5d$exports);


var $4c173a6fdfcd47dd$exports = {};
var $576ef043a8418447$exports = {};


var $97a4f96ae2e58117$exports = {};


var $61bc14c2d15aaac5$exports = {};


var $f9bb85f259f162ce$exports = {};


var $6d1267bec7b1655c$exports = {};


var $4ab9cd3dbaed7a41$exports = {};


var $5a3aa58630e6f219$exports = {};

$parcel$export($5a3aa58630e6f219$exports, "Studio", () => $5a3aa58630e6f219$export$83fb124b13439b08);
/**
 * @export
 * @namespace Studio
 */ var $5a3aa58630e6f219$export$83fb124b13439b08;
(function(Studio) {
    /**
     * @export
     * @enum {string}
     */ let TypeEnum;
    (function(TypeEnum) {
        TypeEnum[TypeEnum["_0"] = "Производство"] = "_0";
        TypeEnum[TypeEnum["_1"] = "Спецэффекты"] = "_1";
        TypeEnum[TypeEnum["_2"] = "Прокат"] = "_2";
        TypeEnum[TypeEnum["_3"] = "Студия дубляжа"] = "_3";
    })(TypeEnum = Studio.TypeEnum || (Studio.TypeEnum = {}));
})($5a3aa58630e6f219$export$83fb124b13439b08 || ($5a3aa58630e6f219$export$83fb124b13439b08 = {}));


var $411f802f189e5809$exports = {};
var $e44922177c2c5ebd$exports = {};


var $b8df8bb23f2a9aa9$exports = {};


var $6c6065826098fa81$exports = {};


var $b6e5862d9d1eb14d$exports = {};


var $35d1eed5028c394b$exports = {};


var $0486ca0a50696bfc$exports = {};


var $189e178c56cf8456$exports = {};


$parcel$exportWildcard($411f802f189e5809$exports, $e44922177c2c5ebd$exports);
$parcel$exportWildcard($411f802f189e5809$exports, $b8df8bb23f2a9aa9$exports);
$parcel$exportWildcard($411f802f189e5809$exports, $6c6065826098fa81$exports);
$parcel$exportWildcard($411f802f189e5809$exports, $b6e5862d9d1eb14d$exports);
$parcel$exportWildcard($411f802f189e5809$exports, $35d1eed5028c394b$exports);
$parcel$exportWildcard($411f802f189e5809$exports, $0486ca0a50696bfc$exports);
$parcel$exportWildcard($411f802f189e5809$exports, $189e178c56cf8456$exports);


$parcel$exportWildcard($4c173a6fdfcd47dd$exports, $576ef043a8418447$exports);
$parcel$exportWildcard($4c173a6fdfcd47dd$exports, $97a4f96ae2e58117$exports);
$parcel$exportWildcard($4c173a6fdfcd47dd$exports, $61bc14c2d15aaac5$exports);
$parcel$exportWildcard($4c173a6fdfcd47dd$exports, $f9bb85f259f162ce$exports);
$parcel$exportWildcard($4c173a6fdfcd47dd$exports, $6d1267bec7b1655c$exports);
$parcel$exportWildcard($4c173a6fdfcd47dd$exports, $4ab9cd3dbaed7a41$exports);
$parcel$exportWildcard($4c173a6fdfcd47dd$exports, $5a3aa58630e6f219$exports);
$parcel$exportWildcard($4c173a6fdfcd47dd$exports, $411f802f189e5809$exports);


var $fd30bbeff255ff31$exports = {};
var $f182492e3ed446f7$exports = {};


var $1d3b6161c5bd7a12$exports = {};


var $7d49ca0871d558a8$exports = {};


var $46ed617f6c3d9d0e$exports = {};


var $63249539923f0b2c$exports = {};


var $33776631f6f71e6a$exports = {};


var $21448869b724b239$exports = {};


$parcel$exportWildcard($fd30bbeff255ff31$exports, $f182492e3ed446f7$exports);
$parcel$exportWildcard($fd30bbeff255ff31$exports, $1d3b6161c5bd7a12$exports);
$parcel$exportWildcard($fd30bbeff255ff31$exports, $7d49ca0871d558a8$exports);
$parcel$exportWildcard($fd30bbeff255ff31$exports, $46ed617f6c3d9d0e$exports);
$parcel$exportWildcard($fd30bbeff255ff31$exports, $63249539923f0b2c$exports);
$parcel$exportWildcard($fd30bbeff255ff31$exports, $33776631f6f71e6a$exports);
$parcel$exportWildcard($fd30bbeff255ff31$exports, $21448869b724b239$exports);




export {$21782052d57b2416$export$d24bb501ea7ded0b as QueryBuilder, $21782052d57b2416$export$819b0b6b1f83fa2a as MovieQueryBuilder, $21782052d57b2416$export$ae3fb5834197d0a9 as PersonQueryBuilder, $21782052d57b2416$export$8659e94a5c99ae98 as ReviewQueryBuilder, $21782052d57b2416$export$1828073978eaf824 as SeasonQueryBuilder, $21782052d57b2416$export$efddcc2335f0c393 as StudioQueryBuilder, $21782052d57b2416$export$a8205ba3f55eed92 as ImageQueryBuilder, $6c79b96500ca857b$export$a40e5455e0e769de as SORT_TYPE, $dd8fe0960d14576e$export$5286ee09df71cffc as SPECIAL_VALUE, $3575132427cc3c5d$export$a4ad2735b021c132 as VERSION, $3c880564bcfe2be9$export$1af66cb2cba0b3db as KinopoiskDev, $5a3aa58630e6f219$export$83fb124b13439b08 as Studio};
